core/ecschema-metadata/src/UnitConversion/UnitConverter.ts

```diff
@@ -0,0 +1,84 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { BentleyError, BentleyStatus } from "@bentley/bentleyjs-core";
+import { Constant, SchemaContext, SchemaItem, SchemaKey, Unit } from "../ecschema-metadata";
+import { UnitConversion } from "./UnitConversion";
+import { UnitGraph } from "./UnitTree";
+
+/**
+ * Class constructed with SchemaContext and used to calculate [[UnitConversion]] between Units
+ * @alpha
+ */
+export class UnitConverter {
+  private _uGraph: UnitGraph;
+
+  /**
+   * Create Converter context
+   * @param _context SchemaContext with contexts added to it.
+   */
+  constructor(private readonly _context: SchemaContext) {
+    this._uGraph = new UnitGraph(this._context);
+  }
+
+  /**
+   * Find conversion between from and to units, formatted {schemaName}.{schemaItemName} or {schemaName}:{schemaItemName}
+   * @param fromFullName SchemaItem full name of source unit
+   * @param toFullName SchemaItem full name of target unit
+   * @returns [[UnitConversion]] converting fromFullName -> toFullName with a factor and an offset
+   * @throws Error if from and to Units' SchemaItem is not found in Schema or Schema prefix is not found in SchemaContext
+   * @throws Error if from and to Units do not belong to the same phenomenon
+   * @throws Error if definitions' SchemaItems cannot be found in its own or referenced Schemas
+   */
+  public async calculateConversion(fromFullName: string, toFullName: string): Promise<UnitConversion> {
```

Comment:
Does this need to be [full name](https://www.itwinjs.org/bis/ec/ec-name/#full-name) or can it be a [qualified name](https://www.itwinjs.org/bis/ec/ec-name/#qualified-name)? In the comment you reference the prefix/alias of the schema.

 I'd suggest changing the params to be `fromUnit` and `toUnit` then in the description qualify which ones you're looking for.

Reply:
Currently, this needs to be the full name. The definitions for the units can be full names or qualified names, but users have to pass in full names for calculateConversion.

Reply:
I agree fromUnit/toUnit is better naming

Reply:
Changed to fromUnit/toUnit